Commit

IOGun opened the pull request:
Added: TOSc, ASeSc, NSeSc, NCSC, NSc, ACSc

IOGun committed:
Commit
Added: TOSc, ASeSc, NSeSc, NCSC, NSc, ACSc

Anna-Malinina commented on src/main/kotlin/ArchiveCreateScreen.kt:
```diff
@@ -0,0 +1,33 @@
+
+class ArchiveCreateScreen(var listOfArchives: MutableList<Archive>): TemplateOfScreen() {
+    override val messagesOfScreen = mutableMapOf(
+            "title_of_screen" to "\n Создание архива ",
```
Можно было бы вынести тег в константу, чтобы при необходимости поменять его, пришлось менять только в одном месте.

Anna-Malinina commented on src/main/kotlin/ArchiveCreateScreen.kt:
```diff
@@ -0,0 +1,33 @@
+
+    override fun createListOfMenu() {
+        listOfMenuItems.clear()
+        listOfMenuItems.add(ItemOfMenu("Создать новый архив", {addArchive(); false} ))
```
Обрати внимание на подсказки студии, она может что-то подчеркивать желтым, а что-то серым (неиспользуемый код или импорты).Комбинация автоматического форматирования поможет убрать неиспользуемые строки, выровнять код: ctrl+alt+L или cmd+opt+L, другие серые и желтые подсказки можно быстро выполнить с помощью комбинации alt+enter или opt+enter. В работе эти две комбинации являются самыми часто используемыми и полезными, так как форматируют и правят за нас код, потому что в идеале не должно быть никаких желтых и серых мест :)

Anna-Malinina commented on src/main/kotlin/ArchiveCreateScreen.kt:
```diff
@@ -0,0 +1,33 @@
+        listOfMenuItems.add(ItemOfMenu("Выход", {exitFromThisScreen(); true }))
+    }
+
+    fun addArchive(): Boolean {
```
Так как этот метод используется только внутри класса, то его можно сделать приватным. Это один из основных [принципов ООП программирования](https://practicum.yandex.ru/blog/obektno-orientirovannoe-programmirovanie/#principy-oop) - инкапсуляция

Anna-Malinina commented on src/main/kotlin/ArchiveCreateScreen.kt:
```diff
@@ -0,0 +1,33 @@
+            if (str.isEmpty()) {
+                println("Ошибка ввода. Название архива не может быть пустым. Добавьте текст.")
+            } else{
+                var title: String = str
```
Тип для каждой переменной можно не указывать явно, компилятор понимает его сам. Так код будет проще)

Anna-Malinina replied:
Эта переменная нигде не меняется, поэтому она должна быть val

Anna-Malinina commented on src/main/kotlin/ArchiveCreateScreen.kt:
```diff
@@ -0,0 +1,33 @@
+                val newArchive : Archive = Archive(title, mutableListOf())
+                listOfArchives.add(newArchive)
+                println("\nАрхив \"${newArchive.name}\" успешно добавлен")
+                return true
```
результат этого метода нигде не используется, при этом он всегда true, что делает его не сильно значимым. Поэтому можно убрать возвращаемый тип у метода, а выйти из него с помощью return без аргументов или просто  break

Anna-Malinina commented on src/main/kotlin/ArchiveSelectionScreen.kt:
```diff
@@ -0,0 +1,24 @@
+
+class ArchiveSelectionScreen(var listOfArchives: MutableList<Archive>) : TemplateOfScreen() {
```
В этом классе аналогичные предложения: поправить форматирование и заприватить все, что может быть приватным)

Anna-Malinina commented on src/main/kotlin/ArchiveSelectionScreen.kt:
```diff
@@ -0,0 +1,24 @@
+   override fun createListOfMenu() {
+        listOfMenuItems.clear()
+        listOfMenuItems.add(ItemOfMenu("Добавить новый архив в список", {openCreateScreen(); false} ))
+        for (row in listOfArchives) {
```
вместо for также можно использовать метод списка listOfArchives.forEach{}

Anna-Malinina commented on src/main/kotlin/Main.kt:
```diff
@@ -1,3 +1,11 @@
+
 fun main(args: Array<String>) {
```
Аргументы можно убрать, если они не используются

Anna-Malinina commented on src/main/kotlin/Main.kt:
```diff
@@ -1,3 +1,11 @@
-    println("Hello World!")
+
+    val listOfArchive: MutableList<Archive> = mutableListOf()
+    val mainScreen: ArchiveSelectionScreen = ArchiveSelectionScreen(listOfArchive)
```
И тоже самое про типы

Anna-Malinina commented on src/main/kotlin/NoteCreateScreen.kt:
```diff
@@ -0,0 +1,34 @@
+import java.util.Scanner
+
+class NoteCreateScreen(var listOfNotes: MutableList<Note>): TemplateOfScreen() {
```
По классу тоже самое, что и по созданию архива

Anna-Malinina commented on src/main/kotlin/NoteScreen.kt:
```diff
@@ -0,0 +1,15 @@
+
+class NoteScreen(val note: Note): TemplateOfScreen() {
```
Так как note не используется нигде, кроме инициализации переменной messagesOfScreen, то можно убрать val, и она будет как аргумент в первичном конструкторе. Val добавляется в конструкторе, если эта переменная используется еще где-то помимо инициализации

Anna-Malinina commented on src/main/kotlin/NoteSelectionScreen.kt:
```diff
@@ -0,0 +1,30 @@
+
+class NoteSelectionScreen(var archive: Archive ): TemplateOfScreen() {
```
Тоже аналогично другим классам)

Anna-Malinina commented on src/main/kotlin/TemplateOfScreen.kt:
```diff
@@ -0,0 +1,76 @@
+
```
Пустые строки перед импортами не ставятся)

Anna-Malinina commented on src/main/kotlin/TemplateOfScreen.kt:
```diff
@@ -0,0 +1,76 @@
+
+data class Note(val text: String)
+
+class ItemOfMenu(val name: String,
```
Здесь полетело форматирование, ctrl +alt + L или cmd + opt + L выровняет красиво)
![image](https://github.com/IOGun/Kotlin-Module-Project/assets/129777932/575e9458-a75b-4f64-9233-48ffacd338ac)

Anna-Malinina commented on src/main/kotlin/TemplateOfScreen.kt:
```diff
@@ -0,0 +1,76 @@
+
+    var listOfMenuItems: MutableList<ItemOfMenu> = mutableListOf()
+
+    open fun createListOfMenu() {
```
Класс можно сделать абстрактным, и этот метод тоже. Тогда не придется писать пустое тело

Anna-Malinina commented on src/main/kotlin/TemplateOfScreen.kt:
```diff
@@ -0,0 +1,76 @@
+        println(messagesOfScreen["title_of_screen"])
+        createListOfMenu()
+        var i: Int = 0
+        for (row in listOfMenuItems){
```
Можно заменить на listOfMenuItems.forEachIndexed { index, row -> } вместо for и кастомного индекса

Anna-Malinina commented on src/main/kotlin/TemplateOfScreen.kt:
```diff
@@ -0,0 +1,76 @@
+    }
+
+    fun checkItemOfMenu(str: String): Int {
+        if (str.toIntOrNull()==null) {
```
Можно return писать перед if, то есть return If () {} else {}, где последняя строка в if и else будет возвращаемым значением

Anna-Malinina commented on src/main/kotlin/TemplateOfScreen.kt:
```diff
@@ -0,0 +1,76 @@
+    }
+
+    fun chooseItemOfMenu(item: Int): Boolean {
+        val result = listOfMenuItems[item].menuFunc.invoke()
```
Можно не выделять в отдельную переменную, а сразу сделать return listOfMenuItems[item].menuFunc.invoke(), или даже fun chooseItemOfMenu(item: Int) = listOfMenuItems[item].menuFunc.invoke()

IOGun committed:
Commit
Изменения:
1) Добавил поле name в Note в TemplateOfScreen.kt
2) Изменил ArchiveCreateScreen.addArchive
3) Изменил по рекомендациям

KorzhMorzh commented:
Только не создавай новую ветку и новый Pr, делай все в dev, коммиты подтянутся в этот пр автоматически, и пришлёшь ссылку на это пр снова

KorzhMorzh commented:
*когда-нибудь я запомню, что на телефоне у меня личный гитхаб, и не буду с него комментарии оставлять*
